agents/agent.py

```diff
@@ -156,14 +140,40 @@ def from_config_file(cls, filename=None, log_path=None):
         Returns:
             ServerAgent: Child instance of ServerAgent.
         """
-        agent = cls(log_path=log_path)
+        agent = cls()
 
         agent._parse_config_file(filename)
+        agent.setup_logging(log_path)
 
         return agent
 
+    def setup_logging(self, log_path=None):
+        """
+        Setup agent's logger based on its server_name.
+
+        Parameters:
+            log_path (PathLike, optional): Path to where logs will be
+                stored.
+        """
+        log_path = (
+            log_path or pkg_resources.resource_filename(
+                self.__class__.__module__, 'logs/%s.log' % self.server_name
+            )
+        )
+
+        logging.config.fileConfig(
+            log_config_path,
+            defaults={
+                'agent_name': self.server_name,
+                'log_path': log_path
+            },
+        )
+
     @property
     def logger(self):
+        if not self.server_name:
```

Comment:
Consider explicitly checking `is None or not self.server_name.strip()` to avoid cases where server_name is a string of whitespace.

Reply:
All your remarks are really valid, I will create a separate issue for them, because this bug is a blocker for some team members.